packages/firestore/src/local/simple_db.ts

```diff
@@ -246,11 +246,8 @@ export class SimpleDbTransaction {
 
   /**
    * A promise that resolves with the result of the IndexedDb transaction.
-   *
-   * Note: A transaction explicitly aborted via abort() is considered successful
-   * and this promise will resolve as successful.
    */
-  readonly completionPromise: Promise<void>;
+  readonly completionPromise: Deferred<void>;
```

Comment:
Err...   exposing the raw Deferred<> as part of the public API of SimpleDbTransaction seems like leaking implementation details. External code shouldn't be able to call `transaction.completionPromise.resolve()` and external code shouldn't have to do `transaction.completionPromise.promise` to get the promise.  I would:

1. Rename completionPromise to completionDeferred and make it private.
2. Add a public completionPromise getter that returns completionDeferred.promise.

Reply:
Done and done.